engines/lab/dispman.cpp

```diff
@@ -0,0 +1,1052 @@
+								*vgaTemp = color;
+
+							vgaTemp++;
+
+							mask = mask >> 1;
+							templeft--;
+						}
+					} else {
+						templeft -= 8;
+						vgaTemp += 8;
+					}
+				}
+
+				vgaTempLine += _screenWidth;
+				leftInSegment -= _screenWidth;
+
+				if (leftInSegment <= 0) {
+					curPage++;
+					vgaTempLine -= _screenBytesPerPage;
+					leftInSegment += _screenBytesPerPage;
+				}
+			}
+		}
+
+		x += tf->_widths[(byte)text[i]];
+	}
+}
+
+void DisplayMan::doScrollBlack() {
+	uint16 width = _vm->_utils->vgaScaleX(320);
+	uint16 height = _vm->_utils->vgaScaleY(149) + _vm->_utils->svgaCord(2);
+	byte *mem = new byte[width * height];
+
+	_vm->_event->mouseHide();
+
+	Image img(_vm);
+	img._width = width;
+	img._height = height;
+	img._imageData = mem;
+	_vm->_music->updateMusic();
+	img.readScreenImage(0, 0);
+	_vm->_music->updateMusic();
+
+	byte *baseAddr = getCurrentDrawingBuffer();
+	uint16 by = _vm->_utils->vgaScaleX(4);
+	uint16 nheight = height;
+
+	while (nheight) {
+		_vm->_music->updateMusic();
+
+		if (!_vm->_isHiRes)
+			_vm->waitTOF();
+
+		baseAddr = getCurrentDrawingBuffer();
+
+		if (by > nheight)
+			by = nheight;
+
+		mem += by * width;
+		nheight -= by;
+		uint32 copySize;
+		uint32 size = (int32)nheight * (int32)width;
+		byte *tempMem = mem;
+
+		while (size) {
+			if (size > _screenBytesPerPage)
+				copySize = _screenBytesPerPage;
+			else
+				copySize = size;
+
+			size -= copySize;
+
+			memcpy(baseAddr, tempMem, copySize);
+			tempMem += copySize;
+		}
+
+		setPen(0);
+		rectFill(0, nheight, width - 1, nheight + by - 1);
+
+		screenUpdate();
+
+		if (!_vm->_isHiRes) {
+			if (nheight <= (height / 8))
+				by = 1;
+			else if (nheight <= (height / 4))
+				by = 2;
+			else if (nheight <= (height / 2))
+				by = 3;
+		}
+	}
+
+	delete[] mem;
+	freePict();
+	_vm->_event->mouseShow();
+}
+
+void DisplayMan::copyPage(uint16 width, uint16 height, uint16 nheight, uint16 startLine, byte *mem) {
+	byte *baseAddr = getCurrentDrawingBuffer();
+
+	uint32 size = (int32)(height - nheight) * (int32)width;
+	mem += startLine * width;
+	uint16 curPage = ((int32)nheight * (int32)width) / _screenBytesPerPage;
+	uint32 offSet = ((int32)nheight * (int32)width) - (curPage * _screenBytesPerPage);
+
+	while (size) {
+		uint32 copySize;
+		if (size > (_screenBytesPerPage - offSet))
+			copySize = _screenBytesPerPage - offSet;
+		else
+			copySize = size;
+
+		size -= copySize;
+
+		memcpy(baseAddr + (offSet >> 2), mem, copySize);
+		mem += copySize;
+		curPage++;
+		offSet = 0;
+	}
+}
+
+void DisplayMan::doScrollWipe(const Common::String filename) {
+	_vm->_event->mouseHide();
+	uint16 width = _vm->_utils->vgaScaleX(320);
+	uint16 height = _vm->_utils->vgaScaleY(149) + _vm->_utils->svgaCord(2);
+
+	while (_vm->_music->isSoundEffectActive()) {
+		_vm->_music->updateMusic();
+		_vm->waitTOF();
+	}
+
+	readPict(filename, true, true);
+	setPalette(_vm->_anim->_diffPalette, 256);
+	byte *mem = _vm->_anim->_rawDiffBM._planes[0];
```

Comment:
It can't be removed, it's in fact some missing code in diffNextFrame()... So it's yet to be fixed